### ISSUE TITLE
PaddleNLP中modeling.py代码疑问

### ISSUE THREAD

**Issue by galaxyknight** [PaddlePaddle/PaddleOCR#9610]
请提供下述完整信息以便快速定位问题/Please provide the following information to quickly locate the problem

- 系统环境/System Environment：CentOS7+Python3.8
- 版本号/Version：Paddle：paddlenlp 2.5.1  PaddleOCR：2.6 问题相关组件/Related components：“python3.8/site-packages/paddlenlp/transformers/layoutxlm/modeling.py”
- 运行指令/Command Code：python  -m paddle.distributed.launch --gpus '1' tools/train.py -c configs/kie/vi_layoutxlm/re_vi_layoutxlm_xfund_zh_udml.yml
- 完整报错/Complete Error Message：
KIE任务中使用到的paddlenlp中"python3.8/site-packages/paddlenlp/transformers/layoutxlm/modeling.py"文件中第1236行这里的shape=[512, 3]是合理的吧，怎么感觉写成shape=[max_seq_len, 3]更合理呢，因为在entities经过处理后会增加一行，1242行取的时候也是从1开始取的。在自己标注的数据集上训练时，会报[513, 3]不等于[512, 3]的维度错。写成shape=[max_seq_len, 3]后就能过。
部分源码如下：
class REDecoder(nn.Layer):
    def __init__(self, hidden_size=768, hidden_dropout_prob=0.1):
        super(REDecoder, self).__init__()
        self.entity_emb = nn.Embedding(3, hidden_size)
        projection = nn.Sequential(
            nn.Linear(hidden_size * 2, hidden_size),
            nn.ReLU(),
            nn.Dropout(hidden_dropout_prob),
            nn.Linear(hidden_size, hidden_size // 2),
            nn.ReLU(),
            nn.Dropout(hidden_dropout_prob),
        )
        self.ffnn_head = copy.deepcopy(projection)
        self.ffnn_tail = copy.deepcopy(projection)
        self.rel_classifier = BiaffineAttention(hidden_size // 2, 2)
        self.loss_fct = CrossEntropyLoss()

    def build_relation(self, relations, entities):
        batch_size, max_seq_len = paddle.shape(entities)[:2]
        new_relations = paddle.full(
            shape=[batch_size, max_seq_len * max_seq_len, 3], fill_value=-1, dtype=relations.dtype
        )
        for b in range(batch_size):
            # ###########################################################
            if entities[b, 0, 0] <= 2:
                entitie_new = paddle.full(shape=[512, 3], fill_value=-1, dtype=entities.dtype)   #  代码疑问处
                entitie_new[0, :] = 2
                entitie_new[1:3, 0] = 0  # start
                entitie_new[1:3, 1] = 1  # end
                entitie_new[1:3, 2] = 0  # label
                entities[b] = entitie_new
            entitie_label = entities[b, 1 : entities[b, 0, 2] + 1, 2]
            # ###########################################################
            all_possible_relations1 = paddle.arange(0, entities[b, 0, 2], dtype=entities.dtype)
            all_possible_relations1 = all_possible_relations1[entitie_label == 1]
            all_possible_relations2 = paddle.arange(0, entities[b, 0, 2], dtype=entities.dtype)
            all_possible_relations2 = all_possible_relations2[entitie_label == 2]

            all_possible_relations = paddle.stack(
                paddle.meshgrid(all_possible_relations1, all_possible_relations2), axis=2
            ).reshape([-1, 2])
            if len(all_possible_relations) == 0:
                all_possible_relations = paddle.full_like(all_possible_relations, fill_value=-1, dtype=entities.dtype)
                all_possible_relations[0, 0] = 0
                all_possible_relations[0, 1] = 1

            relation_head = relations[b, 1 : relations[b, 0, 0] + 1, 0]
            relation_tail = relations[b, 1 : relations[b, 0, 1] + 1, 1]
            positive_relations = paddle.stack([relation_head, relation_tail], axis=1)

            all_possible_relations_repeat = all_possible_relations.unsqueeze(axis=1).tile(
                [1, len(positive_relations), 1]
            )
            positive_relations_repeat = positive_relations.unsqueeze(axis=0).tile([len(all_possible_relations), 1, 1])
            mask = paddle.all(all_possible_relations_repeat == positive_relations_repeat, axis=2)
            negative_mask = paddle.any(mask, axis=1) == False
            negative_relations = all_possible_relations[negative_mask]

            positive_mask = paddle.any(mask, axis=0) == True
            positive_relations = positive_relations[positive_mask]
            if negative_mask.sum() > 0:
                reordered_relations = paddle.concat([positive_relations, negative_relations])
            else:
                reordered_relations = positive_relations

            relation_per_doc_label = paddle.zeros([len(reordered_relations), 1], dtype=reordered_relations.dtype)
            relation_per_doc_label[: len(positive_relations)] = 1
            relation_per_doc = paddle.concat([reordered_relations, relation_per_doc_label], axis=1)
            assert len(relation_per_doc[:, 0]) != 0
            new_relations[b, 0] = paddle.shape(relation_per_doc)[0].astype(new_relations.dtype)
            new_relations[b, 1 : len(relation_per_doc) + 1] = relation_per_doc
            # new_relations.append(relation_per_doc)
        return new_relations, entities
